README.md

```diff
@@ -949,15 +950,222 @@ To make writing a SPA easier, a number of frameworks have been developed. The mo
+
+The web application in our example should have a form with a `epsilon` and `guess` input field and a submit button.
+The form in JSX can be written in the following way:
+
+```{.jsx #react-form}
+<form onSubmit={handleSubmit}>
+  <label>
+    Epsilon:
+    <input name="epsilon" type="number" value={epsilon} onChange={onEpsilonChange}/>
+  </label>
+  <label>
+    Initial guess:
+    <input name="guess" type="number" value={guess} onChange={onGuessChange}/>
+  </label>
+  <input type="submit" value="Submit" />
+</form>
+```
+
+The form tag has a `handleSubmit` variable, which is a function that will handle the form submission.
+The input tag will render the `value` and when the user changes the value the `onChange` function will be called.
+
+Let's implement the value and onChange for the epsilon input.
+To store the value will use the [React useState hook](https://reactjs.org/docs/hooks-state.html).
+
+```{.js #react-state}
+const [epsilon, setEpsilon] = React.useState(0.001);
+```
+
+The argument of the `useState` function is the initial value. The `epsilon` variable contains the current value for epsilon and `setEpsilon` is a function to set epsilon to a new value.
+
+The input tag in the form will call the onChange function with a event object. We need to dig the user supplied new value out of the event and pass it to `setEpsilon`.
+
+```{.js #react-state}
+function onEpsilonChange(event) {
+  setEpsilon(event.target.value);
+}
+```
+
+The guess input will be given the same treatment.
+
+```{.js #react-state}
+const [guess, setGuess] = React.useState(-20);
+
+function onGuessChange(event) {
+  setGuess(event.target.value);
+}
+```
+
+We are ready to implement the `handleSubmit` function.
+The function will get, similar to the onChange of the input tag, an event object.
+Normally when you submit a form the form fields will be send to the server, but we want to perform the calculation in the browser so we have to disable the default action with.
+
+```{.jsx #handle-submit}
+event.preventDefault();
+```
+
+Like we did in the previous chapter we have to construct a web worker.
+
+```{.jsx #handle-submit}
+const worker = new Worker('worker.js');
+```
+
+We have to post a message to the worker with the values from the form.
+
+```{.jsx #handle-submit}
+worker.postMessage({
+  type: 'CALCULATE',
+  data: { epsilon: epsilon, guess: guess }
+});
+```
+
+We need a place to store the resulting `root` value, we will use `useState` function again.
+The initial value is set to `undefined` as the `root` value is only known after the calculation has been completed.
+
+```{.js #react-state}
+const [root, setRoot] = React.useState(undefined);
+```
+
+When the worker is done it will send a message back to the app. The app needs to store the `root` result value. The worker can then be terminated because it did its job.
```

Comment:
```suggestion
When the worker is done it will send a message back to the app. The app needs to store the result value (`root`) using `setRoot`. The worker can then be terminated because it did its job.
```